src/redisearch_rs/rlookup/src/bindings.rs

```diff
@@ -43,3 +57,427 @@ pub enum FieldSpecType {
     Geometry = 32,
 }
 pub type FieldSpecTypes = BitFlags<FieldSpecType>;
+
+/// KeyMode is a set of flags that can be used when opening a key with `RedisModule_OpenKey`.
+/// See https://redis.io/docs/latest/develop/reference/modules/modules-api-ref/#RedisModule_OpenKey
+/// TODO: [MOD-10548] move to unified RedisModule wrapper crate.
+#[bitflags]
+#[repr(u32)] // should be c_unit
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub enum KeyMode {
+    /// Open the key for reading.
+    Read = 1 << 0,
+
+    /// Open the key for writing.
+    Write = 1 << 1,
+
+    /// Avoid touching the LRU/LFU of the key when opened.
+    NoTouch = 1 << 16,
+
+    /// Don't trigger keyspace event on key misses
+    NoNotify = 1 << 17,
+
+    /// Don't update keyspace hits/misses counters.
+    NoStats = 1 << 18,
+
+    /// Avoid deleting lazy expired keys.
+    NoExpire = 1 << 19,
+
+    /// Avoid any effects from fetching the key.
+    NoEffects = 1 << 20,
+
+    /// Allow access to expired keys that haven't been deleted yet.
+    AccessExpired = 1 << 21,
+}
+pub type KeyModes = BitFlags<KeyMode>;
+
+/// KeyTypes is a enum of types of a `RedisModuleKey`.
+/// TODO: [MOD-10548] move to unified RedisModule wrapper crate.
+#[repr(u32)]
+#[derive(Copy, Clone, Debug, PartialEq, strum::FromRepr)]
+pub enum KeyTypes {
+    Empty = 0,
+    String = 1,
+    List = 2,
+    Hash = 3,
+    Set = 4,
+    ZSet = 5,
+    Module = 6,
+    Stream = 7,
+}
+
+/// ReplyTypes is a enum of types that can be encapsulated with a `RedisModuleReply`.
+/// TODO: [MOD-10548] move to unified RedisModule wrapper crate.
+#[repr(u32)] // should be c_unit
+#[derive(Copy, Clone, Debug, PartialEq, strum::FromRepr)]
+pub enum ReplyTypes {
+    Unknown = 0xFFFFFFFF,
+    String = 0,
+    Error = 1,
+    Integer = 2,
+    Array = 3,
+    Null = 4,
+    Map = 5,
+    Set = 6,
+    Bool = 7,
+    Double = 8,
+    BigNumber = 9,
+    VerbatimString = 10,
+    Attributes = 11,
+    Promise = 12,
+}
+
+// ===== RedisString =====
+
+/// A new-type wrapper around `RedisModuleString` that ensures proper memory management.
+/// It is designed to be used with the RAII pattern, where the string is automatically freed when it goes out of scope.
+///
+/// Use [`RedisString::from_raw_parts`] to create an instance from raw parts. The c string is accessible via [`RedisString::as_c_str`].
+/// The methods [`RedisString::to_i64`] and [`RedisString::to_f64`] can be used to convert the string to integers or floating-point numbers, respectively.
+#[derive(Debug)]
+pub struct RedisString {
+    // the context in which the string was created, used for freeing the string
+    ctx: *mut RedisModuleCtx,
```

Comment:
What guarantees this pointer stays valid during the object lifetime?
Should this be documented in the safety pre-conditions?